pandas/core/arrays/arrow/array.py

```diff
@@ -949,7 +951,68 @@ def _indexing_key_to_indices(
             indices = np.arange(n)[key]
         return indices
 
-    # TODO: redefine _rank using pc.rank with pyarrow 9.0
+    def _rank(
+        self: ArrowExtensionArrayT,
+        *,
+        axis: AxisInt = 0,
+        method: str = "average",
+        na_option: str = "keep",
+        ascending: bool = True,
+        pct: bool = False,
+    ) -> ArrowExtensionArrayT:
+        """
+        See Series.rank.__doc__.
+        """
+        if axis != 0:
+            raise NotImplementedError
+
+        if (
+            pa_version_under9p0
+            # as of version 10, pyarrow does not support an "average" method
+            or method not in ("min", "max", "first", "dense")
+        ):
+            from pandas.core.algorithms import rank
+
+            ranked = rank(
+                self.to_numpy(),
+                axis=axis,
+                method=method,
+                na_option=na_option,
+                ascending=ascending,
+                pct=pct,
+            )
+            result = pa.array(ranked, type=pa.float64(), from_pandas=True)
+            return type(self)(result)
+
+        sort_keys = "ascending" if ascending else "descending"
+
+        if na_option == "top":
+            null_placement = "at_start"
+        else:
+            null_placement = "at_end"
+
+        result = pc.rank(
+            self._data.combine_chunks(),
+            sort_keys=sort_keys,
+            null_placement=null_placement,
+            tiebreaker=method,
+        )
+        if not pa.types.is_floating(result.type):
```

Comment:
Is this because the existing rank implementation only returns `float64`. If so, I think it's okay to diverge and return whatever type pyarrow returns e.g. if `rank(int64) -> int64` I think we should respect that

Reply:
Yes, the existing implementation returned float64 (numpy). `pc.rank` returns float64 (pyarrow) I figured we should stay consistent with pyarrow which is what we're doing in other methods (e.g. _quantile)

Reply:
Right, but if `pc.rank` returns `pa.int64` this would convert the result to `pa.float64()`

I'm suggesting we just keep the `pa.int64` result here i.e. remove this check

Reply:
Let me know if that's what you had in mind.

Reply:
I think our comments got crossed and were now on the same page?